lib/devicetrust/authz/authz.go

```diff
@@ -48,7 +48,7 @@ func IsTLSDeviceVerified(ext *tlsca.DeviceExtensions) bool {
 // VerifyTLSUser verifies if the TLS identity has the required extensions to
 // fulfill the device trust configuration.
 func VerifyTLSUser(ctx context.Context, dt *types.DeviceTrust, identity tlsca.Identity) error {
```

Comment:
@strideynet pointed out that there are code paths which call `IsTLSDeviceVerified` directly (so don't currently handle bots correctly).

For example: the Kubernetes proxy, but _only_ when setting `device_trust_mode` at the role level, setting it at the cluster level works fine.

I have a fix but need to write tests, so I'll pick this up again on Monday 🙇🏻‍♂️

Reply:
Fixed in [501bfbe](https://github.com/gravitational/teleport/pull/56907/commits/501bfbe0678a86511c62b53b0db06d041868a516).